### ISSUE TITLE
Fix Request: false positive for MissingEqualsAndHashCodeWithGetterSetter on enum

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#324]
Enum types have a built in equals and hashCode implementation, based on enum instances being unique, equality by identity. No need to generate by Lombok.
example:
```
@RequiredArgsConstructor
@Getter // false positive
public enum DomainContext {

    DK("001"),
    VK("002");

    private final String code;
}
```
